parallel-install/pkg/deployment/deletion.go

```diff
@@ -313,120 +282,53 @@ func (i *Deletion) deleteKymaNamespaces(namespaces []string) error {
-					Resource: "clusterassets",
-				}
 
-				assets, err := i.dClient.Resource(caResource).List(context.Background(), metav1.ListOptions{})
-				if err != nil {
-					errorCh <- err
-				}
-				if assets != nil {
-					for _, asset := range assets.Items {
-						asset.SetFinalizers(nil)
-						_, err := i.dClient.Resource(caResource).Update(context.Background(), &asset, metav1.UpdateOptions{})
-						if err != nil {
+						customResourceList, err := i.dClient.Resource(customResource).Namespace(ns).List(context.Background(), metav1.ListOptions{})
+						if err != nil && !apierr.IsNotFound(err) {
 							errorCh <- err
 						}
-						i.cfg.Log.Infof("Deleted finalizer from Cluster Asset: %s", asset.GetName())
+						if customResourceList != nil {
+							for _, cr := range customResourceList.Items {
+								cr.SetFinalizers(nil)
+								_, err := i.dClient.Resource(customResource).Namespace(ns).Update(context.Background(), &cr, metav1.UpdateOptions{})
+								if err != nil {
+									errorCh <- err
+								}
+								i.cfg.Log.Infof("Deleted finalizer from %s: %s", cr.GetKind(), cr.GetName())
+							}
+						}
 					}
 				}
-
-				//HACK: Delete finalizers of leftover Cluster Buckets
-				cbResource := schema.GroupVersionResource{
-					Group:    "rafter.kyma-project.io",
-					Version:  "v1beta1",
-					Resource: "clusterbuckets",
+			}
+			err := retry.Do(func() error {
+				//remove namespace
+				if err := i.kubeClient.CoreV1().Namespaces().Delete(context.Background(), ns, metav1.DeleteOptions{}); err != nil && !apierr.IsNotFound(err) {
+					errorCh <- err
 				}
 
-				buckets, err := i.dClient.Resource(cbResource).List(context.Background(), metav1.ListOptions{})
-				if err != nil {
+				nsT, err := i.kubeClient.CoreV1().Namespaces().Get(context.Background(), ns, metav1.GetOptions{})
+				if err != nil && !apierr.IsNotFound(err) {
 					errorCh <- err
+				} else if apierr.IsNotFound(err) {
+					return nil
 				}
-				if buckets != nil {
-					for _, bucket := range buckets.Items {
-						bucket.SetFinalizers(nil)
-						_, err := i.dClient.Resource(cbResource).Update(context.Background(), &bucket, metav1.UpdateOptions{})
-						if err != nil {
-							errorCh <- err
-						}
-						i.cfg.Log.Infof("Deleted finalizer from Cluster Bucket: %s", bucket.GetName())
-						err = i.dClient.Resource(cbResource).Delete(context.Background(), bucket.GetName(), metav1.DeleteOptions{})
-						if err != nil {
-							errorCh <- err
-						}
-					}
-				}
-			}
-			//remove namespace
-			if err := i.kubeClient.CoreV1().Namespaces().Delete(context.Background(), ns, metav1.DeleteOptions{}); err != nil && !apierr.IsNotFound(err) {
+				i.cfg.Log.Infof("Namespace '%s' still exists with Phase: '%s'", nsT.Name, nsT.Status.Phase)
+				return fmt.Errorf("Namespace '%s' still exists with Phase: '%s'", nsT.Name, nsT.Status.Phase)
+			}, i.retryOptions...)
+			if err != nil {
 				errorCh <- err
 			}
 			i.cfg.Log.Infof("Namespace '%s' is removed", ns)
```

Comment:
Nothing critical, but in [line 356](https://github.com/kyma-project/hydroform/pull/307/files#diff-b6c702ed36695a7a97c142d9f909c60d1c11144074eec3c3aa7b9f726570abe9R356) shouldn't we use something like multi error instead of wrapping? Won't the errors from different goroutines get mixed up right now?